packages/plugins/i18n/admin/src/components/BulkLocaleActionModal.tsx

```diff
@@ -0,0 +1,258 @@
+import * as React from 'react';
+
+import { Table, useTable } from '@strapi/admin/strapi-admin';
+import { Box, Typography, IconButton, Flex, Tooltip } from '@strapi/design-system';
+import { Pencil, CheckCircle, CrossCircle, ArrowsCounterClockwise } from '@strapi/icons';
+import { DocumentStatus } from '@strapi/plugin-content-manager/strapi-admin';
+import { Modules } from '@strapi/types';
+import { stringify } from 'qs';
+import { type MessageDescriptor, useIntl } from 'react-intl';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { Locale } from '../../../shared/contracts/locales';
+import { getTranslation } from '../utils/getTranslation';
+
+import { LocaleStatus } from './CMHeaderActions';
+
+type status = Modules.Documents.Params.PublicationStatus.Kind | 'modified';
+
+/* -------------------------------------------------------------------------------------------------
+ * EntryValidationText
+ * -----------------------------------------------------------------------------------------------*/
+
+const TypographyMaxWidth = styled(Typography)`
+  max-width: 300px;
+`;
+
+interface EntryValidationTextProps {
+  status: status;
+  validationErrors?: Record<string, MessageDescriptor>;
+}
+
+const isMessageDescriptor = (obj: unknown): obj is MessageDescriptor => {
+  const possibleMessageDescriptor = obj as {
+    id?: unknown;
+    defaultMessage?: unknown;
+  };
+
+  return (
+    typeof obj === 'object' &&
+    typeof possibleMessageDescriptor.id === 'string' &&
+    typeof possibleMessageDescriptor.defaultMessage === 'string'
+  );
+};
```

Comment:
```suggestion
const isMessageDescriptor = (obj: unknown): obj is MessageDescriptor => {
  return (
    typeof obj === 'object' &&
    'id' in obj && typeof obj.id === 'string' &&
    "defaultMessage" in obj && typeof obj.defaultMessage === 'string'
  );
};
```

This is what TS wanted you to do :)